webapi/src/main/scala/org/knora/webapi/store/iiif/SipiConnector.scala

```diff
@@ -232,20 +231,70 @@ class SipiConnector extends Actor with ActorLogging {
         } yield SipiConversionResponseV1(fileValueV1, file_type = fileTypeEnum)
     }
 
+    /**
+     * Represents a response from Sipi's `knora.json` route.
+     *
+     * @param originalFilename the file's original filename, if known.
+     * @param originalMimeType the file's original MIME type.
+     * @param internalMimeType the file's internal MIME type (https://dasch.myjetbrains.com/youtrack/issue/DSP-711).
+     * @param mimeType         the file's internal MIME type (https://dasch.myjetbrains.com/youtrack/issue/DSP-711).
+     * @param width            the file's width in pixels, if applicable.
+     * @param height           the file's height in pixels, if applicable.
+     * @param numpages         the number of pages in the file, if applicable.
+     */
+    case class SipiKnoraJsonResponse(originalFilename: Option[String],
+                                     originalMimeType: Option[String],
+                                     internalMimeType: Option[String],
+                                     mimeType: Option[String],
+                                     width: Option[Int],
+                                     height: Option[Int],
+                                     numpages: Option[Int]) {
+        if (originalFilename.contains("")) {
+            throw SipiException(s"Sipi returned an empty originalFilename")
+        }
+
+        if (originalMimeType.contains("")) {
+            throw SipiException(s"Sipi returned an empty originalMimeType")
+        }
+    }
+
+    object SipiKnoraJsonResponseProtocol extends SprayJsonSupport with DefaultJsonProtocol {
+        implicit val sipiKnoraJsonResponseFormat: RootJsonFormat[SipiKnoraJsonResponse] = jsonFormat7(SipiKnoraJsonResponse)
+    }
+
     /**
      * Asks Sipi for metadata about a file.
      *
      * @param getFileMetadataRequestV2 the request.
      * @return a [[GetFileMetadataResponseV2]] containing the requested metadata.
      */
     private def getFileMetadataV2(getFileMetadataRequestV2: GetFileMetadataRequestV2): Try[GetFileMetadataResponseV2] = {
-        val knoraInfoUrl = getFileMetadataRequestV2.fileUrl + "/knora.json"
+        import SipiKnoraJsonResponseProtocol._
 
-        val request = new HttpGet(knoraInfoUrl)
+        val knoraInfoUrl = getFileMetadataRequestV2.fileUrl + "/knora.json"
+        val sipiRequest = new HttpGet(knoraInfoUrl)
 
         for {
-            responseStr <- doSipiRequest(request)
-        } yield responseStr.parseJson.convertTo[GetFileMetadataResponseV2]
+            sipiResponseStr <- doSipiRequest(sipiRequest)
+            sipiResponse: SipiKnoraJsonResponse = sipiResponseStr.parseJson.convertTo[SipiKnoraJsonResponse]
+
+            // Workaround for https://dasch.myjetbrains.com/youtrack/issue/DSP-711
```

Comment:
@benjamingeer it seems like Lukas has already fixed the issue of DSP-711, see [his response](https://dasch.myjetbrains.com/youtrack/issue/DSP-711#focus=Comments-4-1112.0-0) should this workaround be removed in this PR or later?